pallets/task-scheduler/src/tasks.rs

```diff
@@ -46,10 +44,7 @@ pub trait VerifiableTask<Runtime: Config> {
 	) -> Result<Runtime::TaskCall, TaskError<Self::EvaluationError, Self::SchedulerError>>;
 }
 
-pub trait TaskV2<Runtime: frame_system::Config> {
-	type Call;
-	type EvaluationError;
-	type SchedulerError;
+pub trait TaskV2<Runtime: frame_system::Config, Call, EvaluationError, SchedulerError> {
```

Comment:
Why the change from associated types to generics?

Or does this change back in the next PR?

Reply:
I think having the same associated types both in `TaskV2` and  `VerifiableTask` is not very informative and unhandy to say the least. The pallet's associated type bounds are VerifiableTask's associated types, TaskV2 is just a helper trait that happens to be generic over `VerifiableTask`'s ATs. The difference is subtle, beyond the ATs vs generics argument.

```rust
impl<T: Config> TaskV2<T, CallFor<T>> for UnverifiedCollectedCoins
where
	UnverifiedCollectedCoins: OffchainVerification<T>,
	<UnverifiedCollectedCoins as OffchainVerification<T>>::Output: Into<T::Balance>,
{
	type EvaluationError = VerificationFailureCause;
	type SchedulerError = SchedulerError;

	fn post_verification_call(
		&self,
		deadline: T::BlockNumber,
		id: &T::Hash,
	) -> Result<CallFor<T>, TaskError<Self::EvaluationError, Self::SchedulerError>>
```

vs 
```rust
impl<T: Config> TaskV2<T, CallFor<T>, VerificationFailureCause, SchedulerError>
	for UnverifiedCollectedCoins
where
	UnverifiedCollectedCoins: OffchainVerification<T>,
	<UnverifiedCollectedCoins as OffchainVerification<T>>::Output: Into<T::Balance>,
{

	fn post_verification_call(
		&self,
		deadline: T::BlockNumber,
		id: &T::Hash,
	) -> Result<CallFor<T>, TaskError<VerificationFailureCause, SchedulerError>>
```

But now TaskV2 disambiguation is as ugly as it can be which is an argument strong enough to stick with ATs. I can go either way, you two choose. @atodorov 
```rust
let transfer_id = TaskV2::<T, _,_,_>::to_id(&pending);
```

Reply:
Hmm, yeah I think the disambiguation alone makes AT worth it IMO.

Reply:
Noted, I will revert the changes on the next PR.